ships/0027-image-abstraction-layer.md

```diff
@@ -0,0 +1,232 @@
+the operator about the mirror registry location and authentication methods. Such mirror registry
+could be configured through a Secret called `mirror-registry-config`, this secret would contain
+the following properties:
+
+| Name       | Description                                                                       |
+| -----------| --------------------------------------------------------------------------------- |
+| address    | The mirror registry URL                                                           |
+| username   | Username                                                                          |
+| password   | The password to be used                                                           |
+| token      | The auth token to be used                                                         |
+| insecure   | Allows access to insecure registries if set to "true"                             |
+| repository | If set all images are going to be mirrored inside this registry repository        |
+
+### Image status
+
+The status field for a given Image should contain all its previously imported versions of the
+same image sorted by import time. Follow below the proposed properties found in an Image `.status`
+property and their meaning:
+
+| Name              | Description                                                                |
+| ----------------- | -------------------------------------------------------------------------- |
+| hashReferences    | A list of all imported versions of the Image. Sorted by import time descending (the first item in this list is the last imported version) |
+
+The property `.status.hashReferences` is a list of imported image versions, The operator would
+hold up to twenty five imports per image. Each item in the list is composed of the following
+properties:
+
+| Name           | Description  |
+| -------------- | ------------ |
+| from           | Keeps a reference from where the image got imported |
+| importedAt     | Date and time of the import |
+| imageReference | Where this reference points to (by hash), may point to the mirror registry in case of a mirrored Image |
+
+### ImageImport status
+
+On a ImageImport status, one would be able to find information about the import attempts executed
+by the operator, it looks like:
+
+| Name           | Description  |
+| -------------- | ------------ |
+| hashReference  | Once the import has succeeded this is populated with the data about the image. Properties for hashReference are equal to Image.Status.HashReferences above. This is what is copied into the Image.Status.HashReferences list. |
+| condition      | The current condition for the ImageImport expressed as a single metav1.Condition{} |
+| importAttempts | A list of all attempts made at processing the ImageImport |
+
+The properties for the object `.status.hashReference` are exactly the same as for the objects
```

Comment:
we need a status.condition here.

Reply:
Would you mind elaborate a little bit more on this ?

Reply:
I think I was referring if we need to have a Condition object under the `.status` subresource of the `ImageImport`. I think I am missing something, but lets assume for a minute that this CRD is an object of the type `run to completion`. If this is the case, we need a Condition of the type `Succeeded`, in order to reflect the state of the resource to users. What type of object do we have here? see [docs](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) for more information.

Reply:
We do have a `condition` in the status. What I am not seeing ?

Reply:
right, I was referring to what condition type we are talking about in here.

Reply:
It makes sense to convert the `condition` property to be a slice instead of a single Condition. I have added a comment to this EP with that regards. PTAL.

Reply:
from my side, u can resolve this thread.